### ISSUE TITLE
Full list of emoji

### ISSUE THREAD

**Issue by robn** [rjbs/Text-SlackEmoji#1]
> you know what would be great?  a patch that started by getting the full list of emoji from the Slack API

I went looking for this a little while ago. Unfortunately the API ([emoji.list](https://api.slack.com/methods/emoji.list)) only lists custom ones, not the stock ones.

I found this, which lists lots: http://www.emoji-cheat-sheet.com/. I don't know what their source is, other than pull requests from users. It seems odd to me that so many services would use the same list without there being a canonical source for it; is there a long chain of reverse-engineering here? I'd wonder if they were off by a few here and there, but that possibly doesn't matter for Text-SlackEmoji.

**Comment by rjbs** [comment]
Ah, yes.  I had a look at that when I first touched the irssi code for this, but gave up when I saw that this didn't use actual emoji to display things, and that it included things that are non-emoji, like :neckbeard: (`:neckbeard:`).  (Hey, Git/Hub seems to use about the same…)  I guess that dumping that page into a file and then going through them one by one would be tolerable.

Annoying, too, to find things like :cat: (`:cat:`) actually being the CAT FACE emoji, while :cat2: (`:cat2:`) is CAT.

**Comment by robn** [comment]
Aha! http://unicodey.com/emoji-data/table.htm (via https://github.com/iamcal/emoji-data, via https://api.slack.com/docs/formatting#emoji).

**Comment by robn** [comment]
Even better: https://raw.githubusercontent.com/iamcal/emoji-data/master/emoji.json

**Comment by rjbs** [comment]
I've added a script to generate the emoji.pl file based on that last repo.  It's still not a perfect match, as I feared.

A less annoying example is that some emoji in Slack are not in this mapping, like from `:simple_smile:` to 🙂, the new "slightly smiling" face.

A more annoying example is the entries in that file that are not in slack, like `:man-kiss-woman:`.  I think heterosexual kisses is `:couplekiss:`, which _is_ in the file.  Anyway, some things will now be mapped that aren't mapped by Slack, but I think that's better than missing out.
